test/functional/rpc_psbt.py

```diff
@@ -187,6 +188,21 @@ def run_test(self):
         # Create and fund a raw tx for sending 10 BTC
         psbtx1 = self.nodes[0].walletcreatefundedpsbt([], {self.nodes[2].getnewaddress():10})['psbt']
 
+        self.log.info("Test that max_tx_weight option is sanity checked and fails when expected")
+        dest_arg = [{self.nodes[0].getnewaddress(): 1}]
+        assert_raises_rpc_error(-8, "Invalid parameter, max tx weight must be between 0 and 400000", self.nodes[0].walletcreatefundedpsbt, [], dest_arg, 0, {"max_tx_weight": -1})
+        assert_raises_rpc_error(-8, "Invalid parameter, max tx weight must be between 0 and 400000", self.nodes[0].walletcreatefundedpsbt, [], dest_arg, 0, {"max_tx_weight": 400001})
+
+        self.log.info("Test that a funded PSBT is always faithful to max_tx_weight option")
+        for weight in [0, 1, MIN_STANDARD_TX_NONWITNESS_SIZE * WITNESS_SCALE_FACTOR]:
+            assert_raises_rpc_error(-4, "The inputs size exceeds the maximum weight", self.nodes[0].walletcreatefundedpsbt, [], dest_arg, 0, {"max_tx_weight": weight})
+
+        sendall_dest = [self.nodes[0].getnewaddress()]
+        max_send_psbt_result = self.nodes[0].sendall(sendall_dest, options={"psbt": True})
+        max_send_weight = self.nodes[0].decoderawtransaction(max_send_psbt_result["hex"])['weight']
+        assert_raises_rpc_error(-4, "Requested max tx weight exceeded:", self.nodes[0].sendall, sendall_dest, options={"max_tx_weight": max_send_weight - 1, "psbt": True})
+        self.nodes[0].sendall(sendall_dest, options={"max_tx_weight": max_send_weight + 4, "psbt": True}) # off by 4 sometimes?
```

Comment:
My first thought would be that we are perhaps not counting the empty witness stack of a non-witness input? :thinking:

Reply:
if this is a bit of overcounting, that's probably ok?

Reply:
Mh, it’s a bit of a smell if we cannot explain this.
I considered input counter, empty witness stack, that we might estimate for a high-R signature in a PSBT, because we didn’t sign it, and that we sometimes get a lower `r` value in the signature and therefore one of the two values is a byte shorter. But our default address type would be at least wrapped segwit, so that would only explain a weight difference of 1. I don’t have a good idea at the moment. @achow101: Do you perhaps have an idea why this is happening?

Reply:
would still like a good explanation for this

Reply:
I verified that this behavior happened, intermittently I get the weight of the transaction off by 4 bytes.
I spent sometime debugging but did not figure why yet.

---
Going by this suggestion https://github.com/bitcoin/bitcoin/pull/29264#discussion_r1468635643 the test is now gone in the new PR https://github.com/bitcoin/bitcoin/pull/29523 , I believe the issue is unrelated.

Should I open an issue for this?